packages/core/src/utils.ts

```diff
@@ -815,38 +829,62 @@ export const isEqualType = (
 }
 
 /**
- * Returns the Create2 address of an implementation contract deployed by ChugSplash, which is
- * calculated as a function of the projectName and the corresponding contract's reference name. Note
+ * Returns the Create2 address of a contract deployed by ChugSplash, which is calculated
+ * as a function of the projectName and the corresponding contract's reference name. Note
  * that the contract may not yet be deployed at this address since it's calculated via Create2.
  *
  * @param projectName Name of the ChugSplash project.
  * @param referenceName Reference name of the contract that corresponds to the proxy.
- * @returns Address of the implementation contract.
+ * @param constructorArgs Constructor arguments for the contract.
+ * @param maybeArtifact Contract artifact, or an object containing the artifactPaths and the integration so the artifact can be resolved.
+ * @returns Address of the contract.
  */
-export const getImplAddress = (
+export const getContractAddress = (
   projectName: string,
   referenceName: string,
-  creationCodeWithConstructorArgs: string
+  constructorArgs: UserConfigVariables,
+  maybeArtifact:
+    | ContractArtifact
+    | {
+        integration: Integration
+        artifactPaths: ArtifactPaths
+      }
```

Comment:
I think we can simplify this by adding a `creationCodeWithConstructorArgs` field on the CRE. We can save that for a future PR though

Reply:
Ya know, I thought about doing this, but IMO, I don't think we use this value enough for it to be worth adding to the CRE. I'm open to it though if there are other things it would simplify.

Reply:
We call `getContractAddress` ~7 times in our codebase, so I think it'd be worth it. It'd make this function less janky IMO